File: PIcloud/src/app/models/speaker.ts
export interface Speaker {
  id: number;
  fullName: string;
  title: string;
  company: string;
  bio: string;
  photoUrl: string | null;
  linkedinUrl: string | null;
  eventIds: number[];
}

export interface SpeakerRequest {
  fullName: string;
  title?: string;
  company?: string;
  bio?: string;
  photoUrl?: string;
  linkedinUrl?: string;
}

/** A speaker profile fetched from LinkedIn — not yet persisted in the database. */
export interface SpeakerCandidate {
  fullName: string;
  title?: string | null;
  company?: string | null;
  bio?: string | null;
  photoUrl?: string | null;
  linkedinUrl?: string | null;
}

File: event-pi/src/main/java/org/example/eventpi/model/Speaker.java
package org.example.eventpi.model;

import jakarta.persistence.*;
import lombok.*;
import java.util.List;

@Entity
@Table(name = "speakers")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Speaker {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "full_name", nullable = false, length = 150)
    private String fullName;

    @Column(length = 200)
    private String title;

    @Column(length = 150)
    private String company;

    @Column(columnDefinition = "TEXT")
    private String bio;

    @Column(name = "photo_url")
    private String photoUrl;

    @Column(name = "linkedin_url")
    private String linkedinUrl;

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(
            name = "event_speakers",
            joinColumns = @JoinColumn(name = "speaker_id"),
            inverseJoinColumns = @JoinColumn(name = "event_id")
    )
    private List<Event> events;
}